README.md

```diff
@@ -81,79 +84,138 @@ end
-
-> It is a wonderful introduction to sending and testing emails with Bamboo. It also covers some of the ways that Bamboo helps catch errors, how some of the internals work, and how to format recipients with the Bamboo.Formatter protocol.
-
-[free bamboo screencast from dailydrip]: https://www.dailydrip.com/topics/elixir/drips/bamboo-email
+Bamboo separates the tasks of email creation and email sending. To use Bamboo,
+you'll need to define one or more email modules (email creation), define a
+mailer module (email sending), and provide some configuration.
 
-Bamboo breaks email creation and email sending into two separate modules. This
-is done to make testing easier and to make emails easy to pipe/compose.
+To create emails, define an email module within your application.
 
 ```elixir
-# In your config/config.exs file
-#
-# There may be other adapter specific configuration you need to add.
-# Be sure to check the adapter's docs. For example, Mailgun requires a `domain` key.
-config :my_app, MyApp.Mailer,
-  adapter: Bamboo.MandrillAdapter,
-  api_key: "my_api_key"
-
-# Somewhere in your application
-defmodule MyApp.Mailer do
-  use Bamboo.Mailer, otp_app: :my_app
-end
-
-# Define your emails
+# some/path/within/your/app/email.ex
 defmodule MyApp.Email do
   import Bamboo.Email
 
   def welcome_email do
     new_email(
-      to: "john@gmail.com",
+      to: "john@example.com",
       from: "support@myapp.com",
       subject: "Welcome to the app.",
       html_body: "<strong>Thanks for joining!</strong>",
       text_body: "Thanks for joining!"
     )
-
-    # or pipe using Bamboo.Email functions
-    new_email()
-    |> to("foo@example.com")
-    |> from("me@example.com")
-    |> subject("Welcome!!!")
-    |> html_body("<strong>Welcome</strong>")
-    |> text_body("welcome")
   end
 end
+```
+
+In addition to the keyword syntax above you can also [compose emails using pipes].
 
-# In a controller or some other module
-Email.welcome_email |> Mailer.deliver_now
+To send emails, define a mailer module for your application that `use`s
+Bamboo's mailer.
 
-# You can also deliver emails in the background with Mailer.deliver_later
-Email.welcome_email |> Mailer.deliver_later
+```elixir
+# some/path/within/your/app/mailer.ex
+defmodule MyApp.Mailer do
+  use Bamboo.Mailer, otp_app: :my_app
+end
 ```
 
+Your configuration will need to know your otp application, your mailer module,
+the adapter you are using, and any additional configuration required by the
+adapter itself.
+
+```elixir
+# config/config.exs
+config :my_app, MyApp.Mailer,
+  adapter: Bamboo.MandrillAdapter,
+  api_key: "my_api_key"
+```
+
+_Other adapter-specific configuration may be required. Be sure to check the
+adapter's docs._
+
+Now that you have configured Bamboo and defined your modules, you can deliver
+email in fitting places within your application.
+
+```elixir
+defmodule MyApp.SomeControllerPerhaps
+  # ... snip ...
+  Email.welcome_email()   # Create your email
```

Comment:
Do you think it would be helpful to provide a full function definition to help with copy-pastabilty? 

```elixir
# some/path/within/your/app/user.ex
defmodule MyApp.User do
  def send_welcome_email do
    MyApp.Email.welcome_email()     # compose the email
    |> MyApp.Mailer.deliver_later() # send the email
  end
end
```

Reply:
👍